internal/prometheus/prometheus.go

```diff
@@ -0,0 +1,76 @@
+package prometheus
+
+import (
+	"net/http"
+
+	"github.com/AdguardTeam/AdGuardHome/internal/stats"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Config - Prometheus configuration
+type Config struct {
+	Enabled   bool   `yaml:"enabled"`
+	Namespace string `yaml:"namespace"`
+
+	AnonymizeClientIP bool `yaml:"-"` // anonymize clients' IP addresses - set from DNS config
+}
+
+type Server struct {
+	conf Config
+
+	dnsRequests *prometheus.CounterVec
```

Comment:
These two metrics are kinda not enough to create a good grafana dashboard:)

Any ideas what else could be added?

Reply:
I agree, although I aimed for having smallest possible scope in this PR and presenting the same statistics as the start page in AdGuard.

I'm not sure what other DNS metrics could be added without having too high cardinality labels (e.g. counters per client IPs or target hostnames). Perhaps the list that caused the block/filter could be added as a label?

Other than that, I guess we could add metrics for other parts of the application, e.g. last time filters were updated?

Reply:
Well, we could add some simple stuff like `build_info`, see how it's done in CoreDNS:
```
var buildInfo = promauto.NewGaugeVec(prometheus.GaugeOpts{
	Namespace: plugin.Namespace,
	Name:      "build_info",
	Help:      "A metric with a constant '1' value labeled by version, revision, and goversion from which CoreDNS was built.",
}, []string{"version", "revision", "goversion"})
```

This should also be an example of how to update prom metrics from a different part of the program.

Reply:
As I have personal interest in having this available, I wanted to provide a small suggestion. There's a project here on GitHub called [ebrianne/adguard-exporter](https://github.com/ebrianne/adguard-exporter). I think it can provide some additional ideas on what could be potentially exposed + there's a Grafana dashboard ready for taking (at least as long as the metrics names are compatible) 😊

Reply:
Oh, thank you! We'll mention it in README then